### PR TITLE
Match DEMO, ai, ar and dsp files

### PR DESCRIPTION
**Pull request by Yanis42**
you won't believe which pikmin game helped matching this (also I fixed a misplaced .text in DEMOFont splits)

### CODE REVIEW

File: src/dolphin/demo/DEMOInit.c [hunk 1]
```diff
@@ -0,0 +1,292 @@
+#include "dolphin.h"
+#include "dolphin/demo.h"
+#include "dolphin/gx.h"
+#include "dolphin/pad.h"
+#include "dolphin/vi.h"
+
+extern unsigned char DemoStatEnable; // size: 0x1, address: 0x0
```
Comment (cadmic) [inline src/dolphin/demo/DEMOInit.c:7]: should we remove these address comments? same with `// .data` and `// functions`

File: src/dolphin/demo/DEMOPuts.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    GXSetTexCoordGen(GX_TEXCOORD0, GX_TG_MTX2x4, GX_TG_TEX0, texMtx);
+}
+
+void DEMOSetupScrnSpc(s32 width, s32 height, f32 depth) {
+    Mtx44 pMtx;
+    Mtx mMtx;
+    f32 top;
+
+    // fixes float ordering issue
+    (void)0.0f;
```
Comment (cadmic) [inline src/dolphin/demo/DEMOPuts.c:57]: this isn't necessary right?

File: src/dolphin/demo/DEMOPuts.c [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                GXTexCoord2s16(s + 0x10, t + 0x10);
+                GXPosition3s16(x + (i * 8), y + 8, z);
+                GXTexCoord2s16(s, t + 0x10);
+            }
+            GXEnd();
+            len = 0;
+        }
+
+        string = str;
+        if (c == 0xA) {
```
Comment (cadmic) [inline src/dolphin/demo/DEMOPuts.c:126]: 0xA -> '\n'?

File: src/dolphin/demo/DEMOPuts.c [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    s32 w;
+    s32 len;
+    s32 i;
+
+    str = string;
+    GXClearVtxDesc();
+    GXSetVtxDesc(9, 1);
+    GXSetVtxDesc(0xD, 1);
+    GXSetVtxAttrFmt(0, 9, 1, 3, 0);
+    GXSetVtxAttrFmt(0, 0xD, 1, 3, 1);
```
Comment (cadmic) [inline src/dolphin/demo/DEMOPuts.c:96]: enumify? see e.g. frame.c for GXSetVtxDesc/GXSetVtxAttrFmt/GXBegin

File: src/dolphin/demo/DEMOInit.c (hunk 1 above)
Comment (Yanis42) [reply src/dolphin/demo/DEMOInit.c:7]: yeah probably, they come from 2001 sdk decomp

File: src/dolphin/demo/DEMOPuts.c [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    s32 w;
+    s32 len;
+    s32 i;
+
+    str = string;
+    GXClearVtxDesc();
+    GXSetVtxDesc(GX_VA_POS, GX_DIRECT);
+    GXSetVtxDesc(GX_VA_TEX0, GX_DIRECT);
+    GXSetVtxAttrFmt(GX_VTXFMT0, GX_VA_POS, GX_POS_XYZ, GX_S16, 0);
+    GXSetVtxAttrFmt(GX_VTXFMT0, GX_VA_TEX0, GX_CLR_RGBA, GX_RGBA4, 1);
```
Comment (cadmic) [inline src/dolphin/demo/DEMOPuts.c:90]: GX_CLR_RGBA -> GX_TEX_ST
GX_RGBA4 -> GX_S16

**Pull request merged**